client/src/app/tabs/cmmn/CmmnEditor.js

```diff
@@ -180,12 +201,25 @@ export class CmmnEditor extends CachedComponent {
 
       modeler.lastXML = xml;
 
-      modeler.importXML(xml, (err) => {
-        if (err) {
-          this.handleError({
-            error: err
-          });
+      this.setState({
+        loading: true
+      });
+
+      modeler.importXML(xml, (error, warnings) => {
+
+        if (error) {
+          return this.handleImport(error);
```

Comment:
Should we pass warnings in case there are some, too?

Reply:
💯